evidence/store.go

```diff
@@ -1,222 +0,0 @@
-package evidence
-
-import (
-	"fmt"
-
-	dbm "github.com/tendermint/tm-db"
-
-	"github.com/tendermint/tendermint/types"
-)
-
-/*
-Requirements:
-	- Valid new evidence must be persisted immediately and never forgotten
-	- Uncommitted evidence must be continuously broadcast
-	- Uncommitted evidence has a partial order, the evidence's priority
-
-Impl:
-	- First commit atomically in outqueue, pending, lookup.
-	- Once broadcast, remove from outqueue. No need to sync
-	- Once committed, atomically remove from pending and update lookup.
-
-Schema for indexing evidence (note you need both height and hash to find a piece of evidence):
-
-"evidence-lookup"/<evidence-height>/<evidence-hash> -> Info
-"evidence-outqueue"/<priority>/<evidence-height>/<evidence-hash> -> Info
-"evidence-pending"/<evidence-height>/<evidence-hash> -> Info
-*/
-
-type Info struct {
-	Committed bool
-	Priority  int64
-	Evidence  types.Evidence
-}
-
-const (
-	baseKeyLookup   = "evidence-lookup"   // all evidence
-	baseKeyOutqueue = "evidence-outqueue" // not-yet broadcast
```

Comment:
it's not immediately obvious to me why the new code doesn't deal with the outqueue or broadcasting logic. can you explain a bit more?

Reply:
Sure I can try explain. So previously there were three buckets that you stored evidence in: outqueue, pending and lookup (actually beforehand evidence was initially stored in all three instead of being moved across from one to the next). The outqueue is for evidence to be broadcasted and is basically the evidence list, the pending was the evidence that had been broadcasted and was awaiting to be proposed in a block and lookup was for all evidence including evidence that had been committed.

Reply:
Evidence was moved around by marking it as broadcasted or committed except that there was never anything that marked it as broadcasted i.e. evidence continued to be sent until it was committed. Hence there only needs to be too distinct buckets, those pending to be submitted (and are also being broadcasted) and those that have been committed.

Reply:
Cool. Can you make sure this gets documented in the commit message when it lands on master?

Reply:
I will do - my concern is that if it is approved doesn't it get automatically merged now - I won't have time to squash and clean up the commits